src/metrics/filters/filter_test.go

```diff
@@ -27,6 +27,18 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func TestPrefixCompositeR2Filter(t *testing.T) {
+	id0 := "arachne_failures"
+	id1 := "arachne_failures_by_rack"
+	f, err := newMultiCharSequenceFilter([]byte("arachne_failures,arachne_failures_by_rack"), false)
+	require.NoError(t, err)
+	_, matches1 := f.matches([]byte(id0))
```

Comment:
I think you're looking at an older commit possibly.

"arachne_failures" should match with "arachne-failures" and "arachne_failures_by_rack" should match with "arachne_failures_by_rack" even if it also matches with other patterns in the filter since this pattern is the best match.

In the latest commit I have tests for the length of the value in the returned byte array and with backwards set to true.

Reply:
Gotcha! Yeah was looking at the older commit. Thanks for the update!